crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -4185,6 +4301,89 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn comparison_tuples() -> anyhow::Result<()> {
```

Comment:
So just this week we landed a brand-new test framework, see the tests (as Markdown files) in `crates/red_knot_python_semantic/resources/mdtest` and the documentation in `crates/red_knot_test/README.md`. Ultimately we would like all of these type inference tests to be written in that form instead of as Rust tests like this. It would be great if we could add this new test as a Markdown test instead of here. It would mean adding a new file (let's say `.../mdtest/comparison/tuples.md`) that includes the Python snippets from this test as embedded Python code blocks, and use `reveal_type(...)` and `# revealed: ` assertions for the types.